go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -792,3 +793,44 @@ func (oc *BaseSecondaryNetworkController) getNetworkID() (int, error) {
 	}
 	return *oc.networkID, nil
 }
+
+// buildUDNEgressSNAT is used to build the conditional SNAT required on L3 and L2 UDNs to
+// steer traffic correctly via mp0 when leaving OVN to the host
+func (bsnc *BaseSecondaryNetworkController) buildUDNEgressSNAT(localPodSubnets []*net.IPNet, outputPort string,
+	node *kapi.Node) ([]*nbdb.NAT, error) {
+	if len(localPodSubnets) == 0 {
+		return nil, nil // nothing to do
+	}
+	var snats []*nbdb.NAT
+	var masqIP *udn.MasqueradeIPs
+	var err error
+	networkID, err := bsnc.getNetworkID()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get networkID for network %q: %v", bsnc.GetNetworkName(), err)
+	}
+	dstMac, err := util.ParseNodeManagementPortMACAddresses(node, bsnc.GetNetworkName())
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse mac address annotation for network %s on node %s, err: %v",
+			bsnc.GetNetworkName(), node.Name, err)
+	}
+	extIDs := map[string]string{
+		types.NetworkExternalID:  bsnc.GetNetworkName(),
+		types.TopologyExternalID: bsnc.TopologyType(),
+	}
+	for _, localPodSubnet := range localPodSubnets {
+		if utilnet.IsIPv6CIDR(localPodSubnet) {
+			masqIP, err = udn.AllocateV6MasqueradeIPs(networkID)
+		} else {
+			masqIP, err = udn.AllocateV4MasqueradeIPs(networkID)
+		}
+		if err != nil {
+			return nil, err
+		}
+		if masqIP == nil {
+			return nil, errors.New("masqIPs can not be empty")
```

Comment:
nit, good to add network name in the error.

Reply:
done.

Reply:
yup done.